conanfile.txt

```diff
@@ -0,0 +1,15 @@
+[requires]
+boost/[>=1.74.0]
+grpc/1.65.0
+protobuf/5.27.0
+xtensor/[>=0.24.3]
+abseil/[>=20230125.3]
```

Comment:
- Some of these versions align with those in the Viam C++ SDK top-level `CMakeLists.txt`, but others differ. Is there a rationale?
- Should there be an explicit `xtl` dependency here (see https://viam.atlassian.net/browse/RSDK-8395)
- The docs on `conanfile.txt` say "it cannot be used to create a package". Is that among the things we want to do? Should this be the more general `conanfile.py` from the beginning?

Reply:
Yes I think it's just grpc and protobuf that differ and those were due to the CVE and version compatibility. 

Also yes that's right. It's an easy migration from a `txt` to a simple `py`, I'll just do that now